File: 1431-kids-with-the-greatest-number-of-candies/1431-kids-with-the-greatest-number-of-candies.java
class Solution {
    public List<Boolean> kidsWithCandies(int[] candies, int extraCandies) {
        int maxCandy = Solution.maxCandy(candies);
        List<Boolean> answer = new ArrayList<>();
        for (int candy: candies) {
            answer.add(candy+extraCandies >= maxCandy);
        }
        return answer;
    }

    static public int maxCandy(int[] candies) {
        int maxCandy = 0;
        for (int candy : candies) {
            if (candy >= maxCandy) {
                maxCandy = candy;
            }
        }
        return maxCandy;
    }
}

File: 1431-kids-with-the-greatest-number-of-candies/1431-kids-with-the-greatest-number-of-candies.go
func kidsWithCandies(candies []int, extraCandies int) []bool {
	max := maxCandies(candies)
	answer := make([]bool, len(candies))
	for i, v := range candies {
		answer[i] = v+extraCandies >= max
	}
	return answer
}

func maxCandies(candies []int) int {
	var answer int
	for _, v := range candies {
		if v > answer {
			answer = v
		}
	}
	return answer
}